chips/sam4l/src/gpio.rs

```diff
@@ -448,7 +448,11 @@ impl hil::Controller for GPIOPin {
     type Config = Option<PeripheralFunction>;
 
 
-    fn configure(&self, config: Option<PeripheralFunction>) {
+    fn configure(&self, config: Self::Config) {
```

Comment:
Changed `Option<PeripheralFunction>` to `Self::Config`, as it seemed to be the intention of the `Config` type